storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArray.java

```diff
@@ -189,6 +189,60 @@ public void setInitialCanonicalBlockRoot(final Bytes32 initialCanonicalBlockRoot
     applyToNodes(this::updateBestDescendantOfParent);
   }
 
+  /**
+   * This function is supposed to be called while syncing. It assumes the protoarray to be in a
+   * state where all nodes have been just initialized (see {@link #setInitialCanonicalBlockRoot})
+   * and have a weight of 1 on the canonical chain and 0 on all other chains.
+   *
+   * <p>It may also do the reorg if we enter syncing mode and, while syncing, we decide to sync to a
+   * new chain so that the common ancestor has not received any votes yet.
+   *
+   * <p>The function set all weights to 0 to the reorged chain and set all weights to 1 to the new
+   * chain.
+   */
+  public void reorgWhileSyncing(
```

Comment:
so we just removed weight from old chain and put it to the new chain undoubtely? We need some reasons for this. Could we get real weights? There is no clue new chain is better than old

Reply:
the idea is to make it the chain we want to reorg to as canonical only by 1 vote. So once we start counting actual votes that +1 becomes irrelevant and goes away as we follow the chain (latest imported block won't have that +1 anymore)